### ISSUE TITLE
dof in t-distribution - mh.py

### ISSUE THREAD

**Issue by anetasie** [sherpa/sherpa#447]
dof should be set to 4 in the t-distribution in dmvt() function in mh.py - it will force a long tail in the distribution. Current setting depends on number of fit parameters.

Per conversation with David Van Dyk:

"Using df=4 in the proposal is standard advice, e.g., in Gelman et al’s text, for the reason you cite. Other proposals are valid, but using a smallish df can help with mixing."

**Comment by anetasie** [comment]
dof=4 should also be set in `rmvt()`

**Comment by anetasie** [comment]
This change may improve convergence in the case of high dof. On the other hand I'm not really sure what value is actually passed to `rmvt()`.

**Comment by DougBurke** [comment]
It's unclear to me how to implement this, as what we are really talking about is some function other than `dmvt` and `rmvt` calling this with the correct value. Or do we want to drop the `dof` parameter completely and hard-code it to 4.

There is essentially no documentation in these modules to help me understand this.

**Comment by anetasie** [comment]
ok here is the file. `dof` is input into several functions. It might be 'easier' to update get_draws() to allow a change in the setting of `dof`.
`https://github.com/sherpa/sherpa/blob/master/sherpa/sim/mh.py`

**Comment by anetasie** [comment]
@DougBurke yes, I don't know in how many places this is defined. This is the scaling for the degrees of freedom in the chi2 distribution used to define the t-distribution. So for the pyblocxs it would be in all the places where the t-distribution is defined and used.

**Comment by anetasie** [comment]
@DougBurke Maybe this doc of pyblocxs can help in understanding the code?
https://hea-www.harvard.edu/AstroStat/pyBLoCXS/

**Comment by DougBurke** [comment]
@anetasie - unfortunately that document isn't very much use here....

**Comment by DougBurke** [comment]
I haven't checked `PragBayes` and `FullBayes` but we have

- `sherpa.ui.utils.get_draws` calls `sherpa.sim.MCMC.get_draws`
- the astro version uses `sherpa.astro.sim.MCMC.get_draws` but it is the same here
- `MCMC.get_draws` sets dof to be the number of free parameters in the fit
- the `dof` value is passed through to the sampler (e.g. `sherpa.sim.MH`, `sherpa.sim.MetropolisMH` and - for the astro case - I think the Bayes versions
- for `MH` and `MetropolisMH` the dof value is only used in calls to `rmvt` and `dmvt` (I am pretty sure this is also true for `PragBayes` and `FullBayes` but I haven't checked that code as closely)

So, we COULD just force the `dof` value in the `MH` class to `4` rather than the number of free parameters. This is an easy change and could be over-ridden in a sub-class of `MH` if felt to be necessary. The problem is is that I don't know if any of the calls to `rmvt` / `dmvt` should use the number of free parameters rather than a hard-coded value of 4. I have no idea who could do this check.

As an example, without really understanding the code, I don't know in the following whether we can just use 4 rather than `_dof` in the calls to `dmvt` and `rmvt` or only one of them: 

```
    def mh(self, current):
        """ MH jumping rule """

        # The current proposal is ignored here.
        # MH jumps from the best-fit parameter values at each iteration
        proposal = rmvt(self._mu, self._sigma, self._dof)
        return proposal

    def dmvt(self, x, log=True, norm=False):
        return dmvt(x, self._mu, self._sigma, self._dof, log, norm)
```
